azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -40,81 +41,240 @@
 TRANSPORT_CERT_FILE_NAME = "TransportCert.pem"
 TRANSPORT_PRV_FILE_NAME = "TransportPrivate.pem"
 
-_NUM_GS_FETCH_RETRIES = 6
+_GET_GOAL_STATE_MAX_ATTEMPTS = 6
 
 
 class GoalState(object):
     def __init__(self, wire_client):
         """
         Fetches the goal state using the given wire client.
 
-        __init__ fetches only the goal state itself, not including inner properties such as ExtensionsConfig; to fetch the entire goal state
-        use the fetch_full_goal_state().
+        Fetching the goal state involves several HTTP requests to the WireServer and the HostGAPlugin. There is an initial request to WireServer's goalstate API,
+        which response includes the incarnation, role instance, container ID, role config, and URIs to the rest of the goal state (ExtensionsConfig, Certificates,
+        Remote Access users, etc.). Additional requests are done using those URIs (all of them point to APIs in the WireServer). Additionally, there is a
+        request to the HostGAPlugin for the vmSettings, which determines the goal state for extensions when using the Fast Track pipeline.
+
+        To reduce the number of requests, when possible, create a single instance of GoalState and use the update() method to keep it up to date.
+        """
+        try:
+            self._wire_client = wire_client
+
+            # These "basic" properties come from the initial request to WireServer's goalstate API
+            self._timestamp = None
+            self._incarnation = None
+            self._role_instance_id = None
+            self._role_config_name = None
+            self._container_id = None
+
+            xml_text, xml_doc = GoalState._fetch_goal_state(self._wire_client)
+
+            self._initialize_basic_properties(xml_doc)
+
+            # The goal state for extensions can come from vmSettings when using FastTrack or from extensionsConfig otherwise, self._fetch_extended_goal_state
+            # populates the '_extensions' property.
+            self._extensions = None
```

Comment:
where is this being used?

Reply:
Thanks, good catch! This was meant to be '_extensions_goal_state', but I did not remove the old variable ('_extensions')